dataland-api-key-manager/src/main/kotlin/org/dataland/datalandapikeymanager/api/ApiKeyAPI.kt

```diff
@@ -45,6 +45,30 @@ interface ApiKeyAPI {
         @RequestParam(required = false) daysValid: Int? = null
     ): ResponseEntity<ApiKeyAndMetaInfo>
 
+    @Operation(
+        summary = "Get API key meta info of a specific user.",
+        description = "Gets meta info about the API key status of a user based on the Keycloak user ID."
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved API key status for the user ID.")
+        ]
+    )
+    @GetMapping(
+        value = ["/getApiKeyMetaInfoForUser"],
+        produces = ["application/json"]
+    )
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @SecurityRequirement(name = "default-bearer-auth")
+    @SecurityRequirement(name = "default-oauth")
+    /** A method to get meta information about the API key status of a specific user, like if that user
+     * even has an API key registered, and what the expiry date of that (potential) API key is.
+     * This method is needed by the Frontend to display the API key status to a logged in user.
+     * @param keycloakUserId is the user ID of the user whose API key status should be retrieved
+     * @return API key meta info which includes all the required info for the API key status
+     */
+    fun getApiKeyMetaInfoForUser(keycloakUserId: String): ResponseEntity<ApiKeyMetaInfo>
```

Comment:
shouldn't keycloakUserId be extracted from the token? so - no need for a parameter here.

Reply:
It is done on the client side, but I agree that it makes more sense to derive the keycloakUserId in the backend.
Will start fixing this now.

Reply:
It's derived from the token in the ApiKeyManager-code now, and not on the clients side anymore.